lxd/images.go

```diff
@@ -852,6 +854,17 @@ func imagesGet(d *Daemon, r *http.Request) Response {
 
 var imagesCmd = Command{name: "images", post: imagesPost, untrustedGet: true, get: imagesGet}
 
+func autoUpdateImagesTask(d *Daemon) (task.Func, task.Schedule) {
+	f := func(context.Context) {
+		autoUpdateImages(d)
+	}
+	schedule := func() (time.Duration, error) {
+		interval := daemonConfig["images.auto_update_interval"].GetInt64()
+		return time.Duration(interval) * time.Hour, nil
+	}
```

Comment:
How does that deal with the case where images.auto_update_interval can be set to 0, meaning not to do any updating?

Reply:
In that case this schedule function would return 0 as interval and the task function would not be run. See also the docstring of task.Schedule:

```
// Schedule captures the signature of a schedule function.
//
// It should return the amount of time to wait before triggering the next
// execution of a task function.
//
// If it returns zero, the function does not get run at all.
```